src/main/scala/com/campudus/tableaux/database/DatabaseConnection.scala

```diff
@@ -19,40 +19,51 @@ class DatabaseConnection(verticle: Verticle) {
 
   case class Transaction(msg: Message[JsonObject]) {
 
-    def query(query: String, values: JsonArray): Future[(Transaction, JsonObject)] = transactionHelper(Json.obj(
+    def query(query: String, values: JsonArray): Future[(Transaction, JsonArray)] = transactionHelper(Json.obj(
       "action" -> "prepared",
       "statement" -> query,
-      "values" -> values)) map { r => (Transaction(r), r.body()) }
+      "values" -> values)) flatMap { r => try Future(Transaction(r), resultChecker(r.body())) catch recover() }
 
     def commit(): Future[Unit] = transactionHelper(Json.obj("action" -> "commit")) map { _ => () }
 
     def rollback(): Future[Unit] = transactionHelper(Json.obj("action" -> "rollback")) map { _ => () }
 
-    def recover(): PartialFunction[Throwable, Future[Transaction]] = {
-      case ex: Throwable => rollback() flatMap (_ => Future.failed[Transaction](ex))
+    def recover(): PartialFunction[Throwable, Future[(Transaction, JsonArray)]] = {
+      case ex: Throwable => rollback() flatMap (_ => Future.failed[(Transaction, JsonArray)](ex))
     }
 
-    private def transactionHelper(jsonObj: JsonObject): Future[Message[JsonObject]] = {
+    private def transactionHelper(json: JsonObject): Future[Message[JsonObject]] = {
       val p = Promise[Message[JsonObject]]()
-      msg.replyWithTimeout(jsonObj, DEFAULT_TIMEOUT, {
-        case Success(rep) => p.success(rep)
-        case Failure(ex) =>
-          verticle.logger.error(s"fail in ${jsonObj.getString("action")}", ex)
-          p.failure(ex)
-      }: Try[Message[JsonObject]] => Unit)
+      msg.replyWithTimeout(json, DEFAULT_TIMEOUT, replyHandler(p, json.getString("action")))
       p.future
     }
   }
 
-  def begin(): Future[Transaction] = {
-    val p = Promise[Transaction]()
-    eb.sendWithTimeout(address, Json.obj("action" -> "begin"), DEFAULT_TIMEOUT, {
-      case Success(rep) =>
-        p.success(Transaction(rep))
-      case Failure(ex) =>
-        verticle.logger.error("fail in begin", ex)
-        p.failure(ex)
-    }: Try[Message[JsonObject]] => Unit)
+  def singleQuery(query: String, values: JsonArray): Future[JsonArray] = sendHelper(Json.obj(
+    "action" -> "prepared",
+    "statement" -> query,
+    "values" -> values)) map { msg => resultChecker(msg.body()) }

```

Comment:
I wouldn't expect it to give me an error all the time. If I just want to say "kick that thing", I may not want to have a 404 but just a "success", as the cell/column/row/link/table/whatever I want to delete is not there (anymore). Maybe I want to do something else with the result of that query. In the case below (https://github.com/campudus/tableaux/pull/10/files#r23123024), that check doesn't make sense for example. Yes, for the delete tests it would make sense, but you can easily add another method that does a singleQuery PLUS the `resultChecker` (the name doesn't really tell me what I should expect from it). Does this make sense?